src/argilla/client/datasets.py

```diff
@@ -823,29 +875,93 @@ def _prepare_for_training_with_spark_nlp(self, records: List[Record]) -> "pandas
                 text = record.text
+            elif record.text is None and "text" in record.inputs:
+                text = record.inputs["text"]
+            else:
+                text = ", ".join(f"{key}: {value}" for key, value in record.inputs.items())
 
             spark_nlp_data.append([record.id, text, record.annotation])
 
         return pd.DataFrame(spark_nlp_data, columns=["id", "text", label_name])
 
-    def __all_labels__(self):
+    def _prepare_for_training_with_openai(self, **kwargs) -> "datasets.Dataset":
+        """Prepares the dataset for training using the "openai" framework.
+
+        Args:
+            **kwargs: Specific to the task of the dataset.
+
+        Returns:
+            A pd.DataFrame.
+        """
+        separator = "\n\n###\n\n"
+        whitespace = " "
+        self.__all_labels__()  # verify that all labels are strings
+
+        if len(self._records) <= len(self._SETTINGS.label_schema) * 100:
+            _LOGGER.warning("OpenAI recommends at least 100 examples for training a classification model.")
+
+        jsonl = []
+        for rec in self._records:
+            if rec.annotation is None:
+                continue
+
+            if rec.text is not None:
+                prompt = rec.text
+            elif rec.text is None and "text" in rec.inputs:
+                prompt = rec.inputs["text"]
+            else:
+                prompt = ", ".join(f"{key}: {value}" for key, value in rec.inputs.items())
+            prompt += separator  # needed for better performance
+
+            if self._records[0].multi_label:
+                completion = ", ".join([str(self._SETTINGS.label2id[annotation]) for annotation in rec.annotation])
+            else:
+                completion = str(self._SETTINGS.label2id[rec.annotation])
+
+            jsonl.append({"id": rec.id, "prompt": prompt, "completion": whitespace + completion})
+
+        return jsonl
+
+    def _infer_settings_from_records(self) -> TextClassificationSettings:
         all_labels = set()
         for record in self._records:
             if record.annotation is None:
                 continue
             elif isinstance(record.annotation, str):
                 all_labels.add(record.annotation)
             elif isinstance(record.annotation, list):
-                all_labels.update((tuple(record.annotation)))
+                for label in record.annotation:
+                    all_labels.add(label)
+            else:
+                # this is highly unlikely
+                raise TypeError("Record.annotation contains an unsupported type: {}".format(type(record.annotation)))
+        all_labels = list(all_labels)
+        all_labels.sort()
+
+        _LOGGER.warning(
+            f"""No label schema provided. Using all_labels: TextClassificationSettings({all_labels}). We recommend providing a `TextClassificationSettings()` or setting it `rg.configure_dataset_settings()`/`rg.load_dataset_settings()` to ensure reproducibility."""
```

Comment:
```suggestion
            f"""No label schema provided. Using all_labels: TextClassificationSettings({all_labels}). We recommend providing a `TextClassificationSettings()` or setting `rg.configure_dataset_settings()`/`rg.load_dataset_settings()` to ensure reproducibility."""
```
I think the line is also a bit too long